### ISSUE TITLE
How could we declare a variable or pointer of  dp::thread_pool

### ISSUE THREAD

**Issue by chenscottus** [DeveloperPaul123/thread-pool#6]
How could we declare variable  dp::thread_pool, for example,
  //in a header h
        dp::thread_pool pool_var;
        //Or 
       dp::thread_pool * pool_ptr;
  in a cpp
       pool_var = dp::thread_pool();

       dp::thread_pool pool
       pool_ptr = &pool;

**Comment by DeveloperPaul123** [comment]
Are you asking how to have it be a member variable of a class? Or are you trying to have the thread pool allocated on the stack and then keep a pointer to it? 

If you could post a more complete example to illustrate what you're trying to do that would be helpful.

**Comment by DeveloperPaul123** [comment]
Any updates on this? @chenscottus

**Comment by chenscottus** [comment]
I figure out:
1) for example, passing a dp::thread_pool to a function, I found out I could declare it something like
      void foo(dp::thread_pool<>& pool) { ... }

2) For example,
In a head file - myclass.h, I will declare a pool variable:
    dp::thread_pool<> pool_vars;
In a cpp file - myclass.cpp, I will initialize the pool variable when myclass running:
    pool_vars = dp::thread_pool ; 

-------------
for example, I would like to declare a variable, and initialize it later
     dp::thread_pool pool_vars;
     ...
     //initialize
     ....

---------
I figure out I could do this:
   std::unique_ptr<dp::thread_pool> pool_vars;
   ...
    pool_vars = std::make_unique<dp::thread_pool>();
   ...

And when we want to destroy it:
    pool_vars.reset();

**Comment by DeveloperPaul123** [comment]
Yes the thread pool is non-copyable by design. You can also use `std::move()` to move the threadpool into your class and therefor avoid the use of `std::unique_ptr` and also avoid a heap allocation.

Also note that you don't need to call `reset()` on a `std::unique_ptr` to destroy the object. This will happen automatically in the destructor.
